pkg/os/os.go

```diff
@@ -23,27 +23,28 @@ import (
 	"path/filepath"
 
 	"github.com/containerd/containerd/mount"
-	"github.com/containerd/fifo"
 	"github.com/docker/docker/pkg/symlink"
-	"golang.org/x/net/context"
-	"golang.org/x/sys/unix"
 )
 
 // OS collects system level operations that need to be mocked out
 // during tests.
 type OS interface {
 	MkdirAll(path string, perm os.FileMode) error
 	RemoveAll(path string) error
-	OpenFifo(ctx context.Context, fn string, flag int, perm os.FileMode) (io.ReadWriteCloser, error)
 	Stat(name string) (os.FileInfo, error)
 	ResolveSymbolicLink(name string) (string, error)
 	FollowSymlinkInScope(path, scope string) (string, error)
 	CopyFile(src, dest string, perm os.FileMode) error
 	WriteFile(filename string, data []byte, perm os.FileMode) error
+	Hostname() (string, error)
+}
+
+// UNIX collects unix system level operations that need to be
+// mocked out during tests.
+type UNIX interface {
```

Comment:
Hmm. You didnt like the approach of defining `OS` in a `_unix.go` and in `_windows.go` so that both just use an `OS` interface but that has different methods on the different platforms?

Reply:
Done.